city_scrapers/spiders/losca_City_Planning.py

```diff
@@ -0,0 +1,101 @@
+import re
+
+from city_scrapers_core.constants import BOARD, COMMISSION, NOT_CLASSIFIED
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+from dateutil.parser import parse
+
+
+class LoscaCityPlanningSpider(CityScrapersSpider):
+    name = "losca_City_Planning"
+    agency = "Los Angeles City Planning"
+    timezone = "America/Los_Angeles"
+    start_year = 2024
+    start_urls = [
+        f"https://planning.lacity.gov/dcpapi/meetings/api/all/commissions/{start_year}",
+        f"https://planning.lacity.gov/dcpapi/meetings/api/all/boards/{start_year}",
+        f"https://planning.lacity.gov/dcpapi/meetings/api/all/hearings/{start_year}",
+    ]
+
+    def parse(self, response):
+        """
+        This spider retrieves meetings for the specified `start_year`.
+        Update `start_year` to change the target year.
+        """
+        try:
+            data = response.json()
+            if "Entries" not in data:
+                self.logger.error("Missing 'Entries' field in response")
+                return
+            items = data["Entries"]
+        except ValueError as e:
+            self.logger.error(f"Failed to parse JSON response: {e}")
+            return
+
+        for item in items:
+            if not all(key in item for key in ["Type", "Date", "Address"]):
+                self.logger.warning(
+                    f"Skipping item due to missing required fields: {item}"
+                )
+                continue
+            meeting = Meeting(
+                title=item.get("Type", ""),
+                description=item.get("Note", ""),
+                classification=self._parse_classification(item),
+                start=parse(item.get("Date")),
+                end=None,
+                all_day=False,
+                time_notes="",
+                location=self._parse_location(item),
+                links=self._parse_links(item),
+                source=self._parse_source(response),
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_location(self, item):
+        """Parse or generate location."""
+        pattern = r"(\d{3,4}(?:\s?–\s?\d{3,4})?(?:\s?and\s?\d{3,4})?(?:\s?,?\s?\d{3,4})?\s[\w\s]+\b(?:Road|Drive|Boulevard|Avenue))"  # noqa
+        addresses = re.findall(pattern, item["Address"])
+        address = ", ".join(addresses) if addresses else item.get("Address", "")
+        return {
+            "address": address,
+            "name": item.get("BoardName", ""),
+        }
```

Comment:
+ In this sample response, your regex pattern accidentally strips the first digit character "2" of the address.
```json
// current location output according to the regex pattern
{
  "address": "1101 West Ventura Boulevard, 1515 South Veteran Avenue, 1523 South Veteran Avenue, 1666 North Vermont Avenue, 1642 – 1666 North Vermont Avenue, 4646 – 4650 West Prospect Avenue, 4685 – 4697 West Hollywood Boulevard",
  "name": ""
}
```
+ IMO, you can just simply parse the whole "Address" key from the response. 

+ I also notice that in case of "cancelled" meetings, the "Address" value is usually updated with "canceled" text, so for such case, we just return empty location.

```js
// Multiple locations for "Address"
{
  "Date": "10/24/2024",
  "Type": "City Planning Commission",
  "APC": "Citywide",
  "CaseNumber": "",
  "BoardName": "",
  "Note": "CITY PLANNING COMMISSION\r\nREGULAR MEETING AGENDA",
  "Address": "21101 West Ventura Boulevard, 1515 South Veteran Avenue (1505, 1507, 1509, 1511, 1513, 1517, 1519, 1521, and 1523 South Veteran Avenue), 1666 North Vermont Avenue (1642 – 1666 North Vermont Avenue;\r\n4646 – 4650 West Prospect Avenue; and 4685 – 4697 West Hollywood Boulevard), ",
  "CD": "",
  "Agenda": "77621",
  "AgendaLink": "http://planning.lacity.gov/dcpapi2/meetings/document/77621",
  "Audio": "77677",
  "AudioLink": "http://planning.lacity.gov/dcpapi2/meetings/document/77677",
  "Minutes": "",
  "MinutesLink": "",
  "Hearing": "",
  "HearingLink": "",
  "AddDocs": "68278",
  "AddDocsLink": "http://planning.lacity.gov/dcpapi2/meetings/document/addtldoc/68278",
  "Time": "",
  "Description": "",
  "workshop": ""
}

// Canceled meeting has text "Canceled Meeting Agenda" included in "Address"
{
  "Date": "10/24/2024",
  "Type": "Area Planning Commission",
  "APC": "South Valley",
  "CaseNumber": "",
  "BoardName": "",
  "Note": "South Valley APC Canceled Meeting Agenda ",
  "Address": "South Valley APC Canceled Meeting Agenda ",
  "CD": "",
  "Agenda": "77622",
  "AgendaLink": "http://planning.lacity.gov/dcpapi2/meetings/document/77622",
  "Audio": "",
  "AudioLink": "",
  "Minutes": "",
  "MinutesLink": "",
  "Hearing": "",
  "HearingLink": "",
  "AddDocs": "",
  "AddDocsLink": "",
  "Time": "",
  "Description": "",
  "workshop": ""
}
```